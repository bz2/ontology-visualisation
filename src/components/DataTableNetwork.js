import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { DataTable } from 'primereact/datatable'
import { Column } from 'primereact/column'
import { useEffect } from 'react'
import actions from '../store/actions'
import setDataTableTriplesLabels from '../utils/dataTableNetwork/setDataTableTriplesLabels'

const DataTableNetwork = ({
  dataTableTriples,
  dataTableTriplesWithLabels,
  globalEdgeStyling,
  userDefinedEdgeStyling,
  globalNodeStyling,
  userDefinedNodeStyling,
  updateStoreValue
}) => {
  const { t } = useTranslation()

  useEffect(() => setDataTableTriplesLabels({
    updateStoreValue
  }), [
    dataTableTriples,
    globalEdgeStyling,
    userDefinedEdgeStyling,
    globalNodeStyling,
    userDefinedNodeStyling,
  ])

  const filterPlaceholder = `${t('filter')}...`

  return (
    <div className="p-p-3 datatable-container elevate-view">
      <DataTable
        header={t('availableRelationships')}
        filter
        removableSort
        scrollable
        sortMode="multiple"
        scrollHeight="95%"
        className="p-datatable-responsive p-datatable-striped"
        value={dataTableTriplesWithLabels}
        emptyMessage={t('noTriplesAvailable')}
      >
        <Column
          field="fromLabel"
          filter
          filterPlaceholder={filterPlaceholder}
          filterMatchMode="contains"
          header={t('from')}
          sortable
        />
        <Column
          field="edgeLabel"
          filterMatchMode="contains"
          filter
          filterPlaceholder={filterPlaceholder}
          header={t('predicate')}
          sortable
        />
        <Column
          field="toLabel"
          filter
          filterPlaceholder={filterPlaceholder}
          filterMatchMode="contains"
          header={t('to')}
          sortable
        />
      </DataTable>
    </div>
  )
}

DataTableNetwork.propTypes = {
  dataTableTriples: PropTypes.arrayOf(PropTypes.shape).isRequired,
  dataTableTriplesWithLabels: PropTypes.arrayOf(PropTypes.shape).isRequired,
  updateStoreValue: PropTypes.func.isRequired,
  globalEdgeStyling: PropTypes.shape().isRequired,
  userDefinedEdgeStyling: PropTypes.shape().isRequired,
  globalNodeStyling: PropTypes.shape().isRequired,
  userDefinedNodeStyling: PropTypes.shape().isRequired,
}

const mapToProps = ({
  dataTableTriples,
  globalEdgeStyling,
  userDefinedEdgeStyling,
  globalNodeStyling,
  userDefinedNodeStyling,
  dataTableTriplesWithLabels
}) => ({
  dataTableTriples,
  globalEdgeStyling,
  userDefinedEdgeStyling,
  globalNodeStyling,
  userDefinedNodeStyling,
  dataTableTriplesWithLabels
})

export default connect(
  mapToProps,
  actions
)(DataTableNetwork)
