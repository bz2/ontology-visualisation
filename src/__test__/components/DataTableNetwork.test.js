import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import DataTableNetwork from '../../components/DataTableNetwork'
import {
  EDGE_COLOR,
  EDGE_COLOR_HIGHLIGHTED,
  CLICK_NODE_BACKGROUND,
  HIGHLIGHT_NODE_BORDER,
  HOVER_NODE_BACKGROUND,
  HOVER_NODE_BORDER,
  LABEL_PROPERTY,
  NODE_BACKGROUND,
  NODE_BORDER,
  NODE_DEFAULT_SHAPE,
  NODE_TEXT_COLOR
} from '../../constants/graph'

const setup = () => {
  const props = {
    updateStoreValue: jest.fn(),
    globalEdgeStyling: {
      stylingEdgeLineColor: EDGE_COLOR,
      stylingEdgeLineColorHover: EDGE_COLOR,
      stylingEdgeLineColorHighlight: EDGE_COLOR_HIGHLIGHTED,
      stylingEdgeLineStyle: false,
      stylingEdgeTextColor: EDGE_COLOR,
      stylingEdgeTextSize: 12,
      stylingEdgeTextAlign: 'horizontal',
      stylingEdgeWidth: 1,
      stylingEdgeLength: 250,
    },
    userDefinedEdgeStyling: {
      stylingEdgeLineColor: EDGE_COLOR,
      stylingEdgeLineColorHover: EDGE_COLOR,
      stylingEdgeLineColorHighlight: EDGE_COLOR_HIGHLIGHTED,
      stylingEdgeLineStyle: false,
      stylingEdgeTextColor: EDGE_COLOR,
      stylingEdgeTextSize: 12,
      stylingEdgeTextAlign: 'horizontal',
      stylingEdgeWidth: 1,
      stylingEdgeLength: 250,
    },
    globalNodeStyling: {
      stylingNodeSize: 25,
      stylingNodeBorder: 1,
      stylingNodeTextColor: NODE_TEXT_COLOR,
      stylingNodeBorderSelected: 2,
      stylingNodeBorderColor: NODE_BORDER,
      stylingNodeBackgroundColor: NODE_BACKGROUND,
      stylingNodeHighlightBorderColor: HIGHLIGHT_NODE_BORDER,
      stylingNodeHighlightBackgroundColor: CLICK_NODE_BACKGROUND,
      stylingNodeHoverBackgroundColor: HOVER_NODE_BACKGROUND,
      stylingNodeHoverBorderColor: HOVER_NODE_BORDER,
      stylingNodeShape: NODE_DEFAULT_SHAPE,
      stylingNodeTextFontSize: 12,
      stylingNodeTextFontAlign: 'center',
      stylingNodeCaptionProperty: LABEL_PROPERTY,
    },
    userDefinedNodeStyling: {
      stylingNodeSize: 25,
      stylingNodeBorder: 1,
      stylingNodeTextColor: NODE_TEXT_COLOR,
      stylingNodeBorderSelected: 2,
      stylingNodeBorderColor: NODE_BORDER,
      stylingNodeBackgroundColor: NODE_BACKGROUND,
      stylingNodeHighlightBorderColor: HIGHLIGHT_NODE_BORDER,
      stylingNodeHighlightBackgroundColor: CLICK_NODE_BACKGROUND,
      stylingNodeHoverBackgroundColor: HOVER_NODE_BACKGROUND,
      stylingNodeHoverBorderColor: HOVER_NODE_BORDER,
      stylingNodeShape: NODE_DEFAULT_SHAPE,
      stylingNodeTextFontSize: 12,
      stylingNodeTextFontAlign: 'center',
      stylingNodeCaptionProperty: LABEL_PROPERTY,
    },
    dataTableTriples: [{
      from: '1',
      edge: '12',
      to: '147'
    }, {
      from: '2',
      edge: '21',
      to: '123'
    }],
    dataTableTriplesWithLabels: [{
      fromLabel: 'Road',
      edgeLabel: 'is affected by',
      toLabel: 'Traffic'
    }, {
      fromLabel: 'Traffic',
      edgeLabel: 'found in',
      toLabel: 'Traffic data'
    }]
  }

  const component = shallow(<DataTableNetwork {...props} />)

  return {
    component,
    props
  }
}

describe('DataTableNetwork', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
