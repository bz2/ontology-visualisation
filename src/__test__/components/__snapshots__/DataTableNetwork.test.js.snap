// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableNetwork should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "updateStoreValue": [Function],
    }
  }
  dataTableTriples={
    Array [
      Object {
        "edge": "12",
        "from": "1",
        "to": "147",
      },
      Object {
        "edge": "21",
        "from": "2",
        "to": "123",
      },
    ]
  }
  dataTableTriplesWithLabels={
    Array [
      Object {
        "edgeLabel": "is affected by",
        "fromLabel": "Road",
        "toLabel": "Traffic",
      },
      Object {
        "edgeLabel": "found in",
        "fromLabel": "Traffic",
        "toLabel": "Traffic data",
      },
    ]
  }
  globalEdgeStyling={
    Object {
      "stylingEdgeLength": 250,
      "stylingEdgeLineColor": "#070b11",
      "stylingEdgeLineColorHighlight": "#9c27b0",
      "stylingEdgeLineColorHover": "#070b11",
      "stylingEdgeLineStyle": false,
      "stylingEdgeTextAlign": "horizontal",
      "stylingEdgeTextColor": "#070b11",
      "stylingEdgeTextSize": 12,
      "stylingEdgeWidth": 1,
    }
  }
  globalNodeStyling={
    Object {
      "stylingNodeBackgroundColor": "#adefd1",
      "stylingNodeBorder": 1,
      "stylingNodeBorderColor": "#011e41",
      "stylingNodeBorderSelected": 2,
      "stylingNodeCaptionProperty": "rdfsLabel",
      "stylingNodeHighlightBackgroundColor": "#ffed00",
      "stylingNodeHighlightBorderColor": "#009688",
      "stylingNodeHoverBackgroundColor": "#f2f2f2",
      "stylingNodeHoverBorderColor": "#607d8b",
      "stylingNodeShape": "circle",
      "stylingNodeSize": 25,
      "stylingNodeTextColor": "#000000",
      "stylingNodeTextFontAlign": "center",
      "stylingNodeTextFontSize": 12,
    }
  }
  mapToProps={[Function]}
  updateStoreValue={[MockFunction]}
  userDefinedEdgeStyling={
    Object {
      "stylingEdgeLength": 250,
      "stylingEdgeLineColor": "#070b11",
      "stylingEdgeLineColorHighlight": "#9c27b0",
      "stylingEdgeLineColorHover": "#070b11",
      "stylingEdgeLineStyle": false,
      "stylingEdgeTextAlign": "horizontal",
      "stylingEdgeTextColor": "#070b11",
      "stylingEdgeTextSize": 12,
      "stylingEdgeWidth": 1,
    }
  }
  userDefinedNodeStyling={
    Object {
      "stylingNodeBackgroundColor": "#adefd1",
      "stylingNodeBorder": 1,
      "stylingNodeBorderColor": "#011e41",
      "stylingNodeBorderSelected": 2,
      "stylingNodeCaptionProperty": "rdfsLabel",
      "stylingNodeHighlightBackgroundColor": "#ffed00",
      "stylingNodeHighlightBorderColor": "#009688",
      "stylingNodeHoverBackgroundColor": "#f2f2f2",
      "stylingNodeHoverBorderColor": "#607d8b",
      "stylingNodeShape": "circle",
      "stylingNodeSize": 25,
      "stylingNodeTextColor": "#000000",
      "stylingNodeTextFontAlign": "center",
      "stylingNodeTextFontSize": 12,
    }
  }
>
  <Component />
</Connect>
`;
